pkg/pool-weighted/test/foundry/LBPool.t.sol

```diff
@@ -0,0 +1,699 @@
+        assertEq(returnedEndTime, endTime, "End time should match");
+        assertEq(returnedEndWeights.length, endWeights.length, "End weights length should match");
+        for (uint256 i = 0; i < endWeights.length; i++) {
+            assertEq(returnedEndWeights[i], endWeights[i], "End weight should match");
+        }
+    }
+
+    function testUpdateWeightsGraduallyMinWeightRevert() public {
+        uint256 startTime = block.timestamp + 1 days;
+        uint256 endTime = startTime + 7 days;
+        uint256[] memory endWeights = new uint256[](2);
+        endWeights[0] = 0.0001e18; // 0.01%
+        endWeights[1] = 0.9999e18; // 99.99%
+
+        vm.prank(bob);
+        vm.expectRevert(WeightedPool.MinWeight.selector);
+        LBPool(address(pool)).updateWeightsGradually(startTime, endTime, endWeights);
+    }
+
+    function testUpdateWeightsGraduallyNormalizedWeightInvariantRevert() public {
+        uint256 startTime = block.timestamp + 1 days;
+        uint256 endTime = startTime + 7 days;
+        uint256[] memory endWeights = new uint256[](2);
+        endWeights[0] = 0.6e18; // 60%
+        endWeights[1] = 0.5e18; // 50%
+
+        vm.prank(bob);
+        vm.expectRevert(WeightedPool.NormalizedWeightInvariant.selector);
+        LBPool(address(pool)).updateWeightsGradually(startTime, endTime, endWeights);
+    }
+
+    function testAddLiquidityRouterNotTrusted() public {
+        RouterMock mockRouter = new RouterMock(IVault(address(vault)), weth, permit2);
+
+        uint256[] memory amounts = [TOKEN_AMOUNT, TOKEN_AMOUNT].toMemoryArray();
+
+        vm.startPrank(bob);
+        vm.expectRevert(abi.encodeWithSelector(LBPool.RouterNotTrusted.selector));
+        mockRouter.addLiquidityUnbalanced(address(pool), amounts, 0, false, "");
+        vm.stopPrank();
+    }
+
+    function testInvalidTokenCount() public {
+        IERC20[] memory sortedTokens1 = InputHelpers.sortTokens([address(dai)].toMemoryArray().asIERC20());
+        IERC20[] memory sortedTokens3 = InputHelpers.sortTokens(
+            [address(dai), address(usdc), address(weth)].toMemoryArray().asIERC20()
+        );
+
+        TokenConfig[] memory tokenConfig1 = vault.buildTokenConfig(sortedTokens1);
+        TokenConfig[] memory tokenConfig3 = vault.buildTokenConfig(sortedTokens3);
+
+        // Attempt to create a pool with 1 token
+        // Doesn't throw InputHelpers.InputLengthMismatch.selector b/c create3 intercepts error
+        vm.expectRevert(Create2.Create2FailedDeployment.selector);
+        LBPoolFactory(address(factory)).create(
+            "Invalid Pool 1",
+            "IP1",
+            tokenConfig1,
+            [uint256(1e18)].toMemoryArray(),
+            DEFAULT_SWAP_FEE,
+            bob,
+            true,
+            ZERO_BYTES32
+        );
+
+        // Attempt to create a pool with 3 tokens
+        // Doesn't throw InputHelpers.InputLengthMismatch.selector b/c create3 intercepts error
```

Comment:
```suggestion
        // Doesn't throw InputHelpers.InputLengthMismatch.selector b/c create2 intercepts error
```

Reply:
I'd consider adding a check at the factory tbh.

Create2 errors are super painful to debug.